test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java

```diff
@@ -12,43 +12,36 @@
 
 public class TestHTTPResourceManager {
 
-    static final String uri;
-    static final Map<Class<?>, TestHTTPResourceProvider<?>> providers;
-
     static {
-        Map<Class<?>, TestHTTPResourceProvider<?>> map = new HashMap<>();
-        for (TestHTTPResourceProvider i : ServiceLoader.load(TestHTTPResourceProvider.class)) {
-            map.put(i.getProvidedType(), i);
-        }
-        providers = Collections.unmodifiableMap(map);
-        Config config = ConfigProvider.getConfig();
-        String host = config.getOptionalValue("quarkus.http.host", String.class).orElse("localhost");
-        String port = config.getOptionalValue("quarkus.http.test-port", String.class).orElse("8081");
-        uri = "http://" + host + ":" + port;
-        System.setProperty("test.url", uri);
+        System.setProperty("test.url", getUri());
```

Comment:
I'm not sure about this line. It could still be subject to the bug observed while injecting the URL...